org.eclipse.transformer/src/main/java/org/eclipse/transformer/Transformer.java

```diff
@@ -945,6 +950,33 @@ public boolean setRules() throws IOException, URISyntaxException, IllegalArgumen
 				dual_info("Java direct string updates will not be performed");
 			}
 
+                    if (!perClassConstantProperties.isEmpty()) {
+                        String masterDirect = getOptionValue(AppOption.RULES_PER_CLASS_CONSTANT, DO_NORMALIZE);
+
+                        Map<String, String> substitutionRefs
+                                = TransformProperties.convertPropertiesToMap(perClassConstantProperties); // throws IllegalArgumentException
+
+                        Map<String, Map<String, String>> masterUpdates = new HashMap<String, Map<String, String>>();
+                        for (Map.Entry<String, String> substitutionRefEntry : substitutionRefs.entrySet()) {
+                            String classSelector = substitutionRefEntry.getKey();
+                            String substitutionsRef = FileUtils.normalize(substitutionRefEntry.getValue());
+
+                            UTF8Properties substitutions = new UTF8Properties();
+                            if (masterDirect == null) {
+                                substitutions = loadInternalProperties("Substitions matching [ " + classSelector + " ]", substitutionsRef);
+                            }
+                            Map<String, String> substitutionsMap
+                                    = TransformProperties.convertPropertiesToMap(substitutions); // throws IllegalArgumentException
+                            masterUpdates.put(classSelector, substitutionsMap);
+                        }
+
+                        perClassConstantStrings = masterUpdates;
+                        dual_info("Per class constant mapping files is enabled");
+
+                    } else {
+                        perClassConstantStrings = null;
+                        dual_info("Per class constant mapping is not enabled");
```

Comment:
Should be "Per class constant mapping files are not enabled"

Reply:
Thanks, done.